src/hooks/useRedeem.ts

```diff
@@ -21,12 +21,18 @@ export function useRedeem() {
       const kpiOptionsToken = getKpiTokenContract(signer, chainId);
       const emp = getKpiOptionsEMP(signer, chainId);
 
-      const approveTx = await kpiOptionsToken.approve(
-        emp.address,
-        ethers.utils.parseEther("20000000000000000000")
-      );
-      addTransaction({ ...approveTx, label: "approve" });
-      await approveTx.wait();
+      const allowance = await kpiOptionsToken.allowance(account, emp.address);
+      const balance = await kpiOptionsToken.balanceOf(account);
+      const hasToApprove = allowance.lt(balance);
+      if (hasToApprove) {
+        const approveTx = await kpiOptionsToken.approve(
+          emp.address,
+          ethers.utils.parseEther("20000000000000000000")
```

Comment:
What's this number, `20000000000000000000`? Some other allowance amount options:
- Maximum `uint`: [This value can be found here](https://github.com/UMAprotocol/protocol/blob/master/packages/common/src/Constants.js#L30) and sets the user's allowance to the maximum. This is arguably invasive as the user might unknowingly approve much more balance than they are comfortable with, but the benefit is that if the user trusts our dApp, then they never have to approve again.
- Just approve the `balance` amount. This is the "safest" for the user but they will have to send another approval before the next transfer

Reply:
Great point, the number was a lazy replacement for `INFINITE_AMOUNT`, I think option 1 is better from a UX perspective, and I think its fine as the options only value after expiry is to be redeemed for UMA, but when approving something like a collateral type for minting tokens (ex umaverse LSP interface), then 2 would be best.
TLDR, going to change `200....0000` to the max `uint` and set it as a variable instead of an hardcoded string